src/clr-addons/datagrid/enum-filter/enum-filter.component.ts

```diff
@@ -117,10 +144,25 @@ export class ClrEnumFilterComponent<T extends { [key: string]: any }>
   }
 
   private getDisplayValues(entries: FilterValue[]): string[] {
-    return entries.map(entry => entry.displayValue);
+    return entries.map(entry => (entry.displayValue === this.emptyValue ? '' : entry.displayValue));
   }
 
   private getValues(entries: FilterValue[]): string[] {
-    return entries.map(entry => entry.value);
+    return entries.map(entry => (entry.value === this.emptyValue ? '' : entry.value));
+  }
+
+  private mapValues(values: (string | FilterValue)[]): FilterValue[] {
+    if (values === null || values.length === 0) {
+      return [];
+    }
+    return values.map(value => this.mapValue(value));
+  }
+
+  private mapValue(value: string | FilterValue): FilterValue {
+    if (value instanceof Object) {
+      return value.value === this.emptyDefaultValue ? { value: this.emptyValue, displayValue: this.emptyValue } : value;
```

Comment:
why do we need to change the "value" at all (I guess it could stay empty aka ''), shouldn't the "displayValue" be enough?
as we here set it to '(Empty)', just to reset it to '' when getting the values